### DISCUSSION TITLE
how to keep this \r after sanitizer.Sanitize

### DISCUSSION THREAD

**Discussion by karangiriKG** [mganss/HtmlSanitizer#343] [Q&A]
<html>
<body>

<h1>HTML DOM Events</h1>
<h2>The onclick Event</h2>

<p>The onclick event triggers a function when an element is clicked on.</p>
<p>Click to trigger a function that will output "Hello World":</p>

</body>
</html>

"%26lt%3bhtml%26gt%3b%3cbr /%3e\r\n%26lt%3bbody%26gt%3b%3cbr /%3e\r\n%3cbr /%3e\r\n%26lt%3bh1%26gt%3bHTML DOM Events%26lt%3b/h1%26gt%3b%3cbr /%3e\r\n%26lt%3bh2%26gt%3bThe onclick Event%26lt%3b/h2%26gt%3b%3cbr /%3e\r\n%3cbr /%3e\r\n%26lt%3bp%26gt%3bThe onclick event triggers a function when an element is clicked on.%26lt%3b/p%26gt%3b%3cbr /%3e\r\n%26lt%3bp%26gt%3bClick to trigger a function that will output %22Hello World%22%3a%26lt%3b/p%26gt%3b%3cbr /%3e\r\n%3cbr /%3e\r\n%3cbr /%3e\r\n%3cbr /%3e\r\n%26lt%3b/body%26gt%3b%3cbr /%3e\r\n%26lt%3b/html%26gt%3b"
While sanitizer.Sanitize(AboveString) it's remove \r 
how to keep this \r after sanitizer.Sanitize

**Comment by mganss** [comment]
There is no immediate way I can think of. The HTML parser does [newline normalization](https://html.spec.whatwg.org/multipage/parsing.html#preprocessing-the-input-stream), eliminating all CRs.

If you want all newlines to be CRLF then of course you can normalize the resulting string after sanitizing.

What's the use case?
